llvm/lib/CodeGen/CodeGenPrepare.cpp

```diff
@@ -8371,8 +8427,10 @@ bool CodeGenPrepare::placeDbgValues(Function &F) {
   for (BasicBlock &BB : F) {
     for (Instruction &Insn : llvm::make_early_inc_range(BB)) {
       DbgValueInst *DVI = dyn_cast<DbgValueInst>(&Insn);
-      if (!DVI)
+      if (!DVI) {
+        MadeChange |= placeDPValues(Insn, DT);
```

Comment:
Shouldn't `placeDPValues` be placed before that `if (!DVI)` check, because a `DbgValueInst` can have `DPValues` attached. It's possibly an error if we have both DP- and dbg-values, so maybe it doesn't matter.

Reply:
Good question -- IMO DPValues on a DbgValueInst isn't a case we should consider as it indicates that something has gone horribly wrong already (and I think there's an assertion in the basic-block converter to catch that?). Were it a DbgDeclare that'd be different as we're supporting those in parallel right now.